experimental/casemapping/src/internals.rs

```diff
@@ -575,6 +641,16 @@ impl<'a> FullMappingResult<'a> {
     }
 }
 
+impl Writeable for FullMappingResult<'_> {
+    fn write_to<W: fmt::Write + ?Sized>(&self, sink: &mut W) -> fmt::Result {
+        match self {
+            FullMappingResult::CodePoint(c) => sink.write_char(*c),
```

Comment:
nit: I prefer matching on refs over derefing, as that explicitly only works on refs.

```suggestion
            FullMappingResult::CodePoint(&c) => sink.write_char(c),
```

Reply:
It's not a ref

Reply:
what is it? we don't get the `char` through a `Deref` impl

Reply:
Just a `char`. It's match arm ergonomics automatically creating a ref. I don't know why it feels the need to do that for a Copy type but whatever.

Reply:
So it is a ref